packages/aws-cdk-lib/aws-ec2/lib/private/ebs-util.ts

```diff
@@ -25,7 +25,30 @@ function synthesizeBlockDeviceMappings<RT, NDT>(construct: Construct, blockDevic
 
     if (ebs) {
 
-      const { iops, volumeType, kmsKey, ...rest } = ebs;
+      const { iops, throughput, volumeType, kmsKey, ...rest } = ebs;
+
+      if (throughput) {
+        const throughputRange = { Min: 125, Max: 1000 };
+        const { Min, Max } = throughputRange;
+
+        if (volumeType != EbsDeviceVolumeType.GP3) {
+          throw new Error('throughput property requires volumeType: EbsDeviceVolumeType.GP3');
+        }
+
+        if (throughput < Min || throughput > Max) {
+          throw new Error(
+            `throughput property takes a minimum of ${Min} and a maximum of ${Max}`,
+          );
+        }
+
+        const maximumThroughputRatio = 0.25;
+        if (iops) {
+          const iopsRatio = (throughput / iops);
+          if (iopsRatio > maximumThroughputRatio) {
+            throw new Error(`Throughput (MiBps) to iops ratio of ${iopsRatio} is too high; maximum is ${maximumThroughputRatio} MiBps per iops`);
```

Comment:
Can you provide documentation about this rule?

Reply:
I cargo culted it from https://github.com/aws/aws-cdk/pull/22441, but after some struggle I did manage to find the documentation that this check is seemingly based on [here](https://docs.aws.amazon.com/ebs/latest/userguide/general-purpose.html#gp3-performance).
> gp3 volumes deliver a consistent baseline throughput performance of 125 MiB/s, which is included with the price of storage. You can provision additional throughput (up to a maximum of 1,000 MiB/s) for an additional cost at a ratio of 0.25 MiB/s per provisioned IOPS. Maximum throughput can be provisioned at 4,000 IOPS or higher and 8 GiB or larger (4,000 IOPS × 0.25 MiB/s per IOPS = 1,000 MiB/s).

Do you want this documentation mentioned somewhere in code?

Reply:
Thanks 👍 Yes I'd update the comment to the variable to specify the behavior (see other comment).